primitives/src/tests.rs

```diff
@@ -7,6 +7,20 @@ use super::{CurrencyId, *};
 use crate::CurrencyInfo;
 use std::str::FromStr;
 
+#[test]
+fn test_trailing_non_alphanum() {
+	let test = b"ZerO";
+	assert_eq!(remove_trailing_non_alphanum_bytes(test).len(), test.len());
+
+	let test = b"Zer0";
+	assert_eq!(remove_trailing_non_alphanum_bytes(test).len(), test.len());
+
+	let test = b"Zer\0";
+	assert_ne!(remove_trailing_non_alphanum_bytes(test).len(), test.len());
```

Comment:
Nit: I was confused at first because the 'assert_**ne**' is easy to miss so you could have used the same structure as in l.22. 
But it really does not matter too much 😅

Reply:
changed